vortex-array/src/patches.rs

```diff
@@ -396,6 +379,89 @@ impl Patches {
     }
 }
 
+/// Filter patches with the provided mask (in flattened space).
+///
+/// The filter mask may contain indices that are non-patched. The return value of this function
+/// is a new set of `Patches` with the indices relative to the provided `mask` rank, and the
+/// patch values.
+fn filter_patches_with_mask<T: ToPrimitive + Copy + Ord>(
```

Comment:
Am I right in thinking the behaviour is this?

```
indices: [0, 2, 3, 5]
mask: [0, 3]

=> [0, 3]
```

As in, isn't it a straight intersection?

In either case, it might nice to do `FilterMask::from_indices(indices).intersection(mask)` and optimize that implementation instead?

Reply:
It's a bit trickier than that, since you also need to track the position within the true_count of the mask so you can build a mask over patch_values.

So you need to simultaneously build an intersection and the result [0, 2] (indices _into_ the indices)

We could also potentially do this as two loops: a loop to find the intersection and a loop to find the mask positions.

Reply:
Ahhhhhh yeah, we had the same thing in Spiral called "intersection_with_indices" or something weird.

Reply:
Do you want me to move this somewhere else or is it ok here? I don't think we do this operation elsewhere in the code currently